plasma_framework/contracts/src/utils/PosLib.sol

```diff
@@ -0,0 +1,78 @@
+pragma solidity 0.5.11;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @dev UTXO position = (blockNumber * BLOCK_OFFSET + txIndex * TX_OFFSET + outputIndex).
+ * TX position = (blockNumber * BLOCK_OFFSET + txIndex * TX_OFFSET)
+ */
+library PosLib {
+    using SafeMath for uint256;
+
+    struct Position {
+        uint256 blockNum;
+        uint256 txIndex;
+        uint16 outputIndex;
+    }
+
+    uint256 constant internal BLOCK_OFFSET = 1000000000;
+    uint256 constant internal TX_OFFSET = 10000;
+
+    /**
+     * @notice Returns the UTXO position for a given transaction position and output index
+     * @param txPos Transaction position - utxo position of zero index output
+     * @param outputIndex Index of the output
+     * @return utxo position
+     */
+    function buildPositionFromTxPosAndOutputIndex(uint256 txPos, uint16 outputIndex)
+        internal
+        pure
+        returns (Position memory)
+    {
+        require(txPos % TX_OFFSET == 0, "Invalid transaction position");
+        return decode(txPos.add(outputIndex));
+    }
+
+    /**
+     * @notice Used for calculating exit priority
+     * @param pos UTXO position for the output
+     * @return Identifier of the transaction
+     */
+    function getTxPostionForExitPriority(Position memory pos)
+        internal
+        pure
+        returns (uint256)
+    {
+        return encode(pos) / TX_OFFSET;
+    }
+
+    /**
+     * @notice Encodes a position
+     * @param pos Position
+     * @return Position encoded as an integer
+     */
+    function encode(Position memory pos) internal pure returns (uint256) {
+        return pos.blockNum * BLOCK_OFFSET + pos.txIndex * TX_OFFSET + pos.outputIndex;
+    }
+
+    /**
+     * @notice Encodes a transaction position
+     * @param pos Transaction position
+     * @return Position encoded as an integer
+     */
+    function encodePackedTxPos(Position memory pos) internal pure returns (uint256) {
```

Comment:
ah...I was actually proposing using this function `encodePackedTxPos` to replace the `getTxPostionForExitPriority` function, so we don't feel like providing function so specifically to exit priority. As it would better be the lib for exit priority to know how it want the encoded bytes IMO.

Also, I was thinking to use `Position txPos` for all internal communications such as `spending condition`, `MoreVpFinalization` so that we don't ever need to specifically encode to a txPos with this schema... however, do realize that there is a gap from our original data is `utxoPos` and need to drop the `outputIndex` field.

Personally I would probably have a helper function to drop the `outputIndex` out, probably sth like:
 `function toStrictTxPos(PosLib.Position utxoPos) returns PosLib.Position`.

I will leave this to you. However, if we are keeping current style, I think this function can be named `encodeTxPos` directly?

Reply:
done